File: stepik-java-api/src/main/java/org/stepik/api/objects/submissions/Submissions.java
package org.stepik.api.objects.submissions;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.objects.ObjectsContainer;

import java.util.ArrayList;
import java.util.List;

/**
 * @author meanmail
 */
public class Submissions extends ObjectsContainer<Submission> {
    private List<Submission> submissions;

    @NotNull
    public List<Submission> getSubmissions() {
        if (submissions == null) {
            submissions = new ArrayList<>();
        }
        return submissions;
    }

    @NotNull
    @Override
    public List<Submission> getItems() {
        return getSubmissions();
    }

    @NotNull
    @Override
    public Class<Submission> getItemClass() {
        return Submission.class;
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/submissions/Submissions.kt
package org.stepik.api.objects.submissions

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import org.stepik.api.client.serialization.DefaultAsEmptySubmissionArrayAdapter
import org.stepik.api.objects.ObjectsContainer


class Submissions : ObjectsContainer<Submission>() {
    @SerializedName("submissions")
    @JsonAdapter(DefaultAsEmptySubmissionArrayAdapter::class, nullSafe = false)
    override var items = mutableListOf<Submission>()

    override val itemClass
        get() = Submission::class.java
}
